pkg/controller/druid/handler.go

```diff
@@ -328,6 +339,44 @@ func deployDruidCluster(sdk client.Client, m *v1alpha1.Druid) error {
 	return nil
 }
 
+func checkCrashStatus(sdk client.Client, m *v1alpha1.Druid) {
+	podList := podList()
+	listOpts := []client.ListOption{
+		client.InNamespace(m.Namespace),
+	}
+	if err := sdk.List(context.TODO(), podList, listOpts...); err != nil {
+		e := fmt.Errorf("failed to list pods for [%s:%s] due to [%s]", m.Kind, m.Name, err.Error())
+		sendEvent(sdk, m, v1.EventTypeWarning, "LIST_FAIL", e.Error())
+		logger.Error(e, e.Error(), "name", m.Name, "namespace", m.Namespace)
+	}
+
+	pod := make([]*v1.Pod, 0)
+	for i := range podList.Items {
+		pod = append(pod, &podList.Items[i])
+	}
+
+	for _, p := range pod {
+		if p.Status.ContainerStatuses[0].RestartCount > 1 {
+			for _, condition := range p.Status.Conditions {
+				if condition.Type == "Ready" {
+					if p.Status.Phase != "Running" || condition.Status == "False" {
```

Comment:
![image](https://user-images.githubusercontent.com/34169002/96025024-36009300-0e72-11eb-9084-cc19dfe6dfb2.png)

Here if you see is the above image is the exact status for crashloopbackOff
1 . incase you are referring to get a simple way to match the condition, then i could go this route
```
 if  if.Status.ContainerStatuses[0].state.waiting.reason == "CrashLoopBackOff" {
                sdkdelete()
}
```
- Why i dint go with the above is due to mainly these reasons : 

1. The ```waiting.reason``` is a string() , reason can be either ErrorImagePull or ImagePullBackOff  or CrashloopackOff or something else. In all sceanrios sts pod  will hangup and all require deletion manually.  Their can be n cases to match.
Plus pod does not directly go to crashloop it will first through error then go to crashstatus, so this string changes.

 Plus if you see the below image thats a imagePullback 
![image](https://user-images.githubusercontent.com/34169002/96025777-3f3e2f80-0e73-11eb-895c-11cabeeeae42.png)

2. Container Restart Count needs to be checked because during initialisation of a pod, the status condition is same, so reconcile loop will keeping on killing it, so in case its > 1 then we match the condition that is it a crashloop ,imagepull, error etc and then delete it. 

3. We are not only dealing with crashloopback, ideally any situation whether error, crashloopback, imagepullback etc which can lead up to hangup state of pod will need to be deleted, with condition.Type and status.Phase we can reach all off them regardless of whatever msg k8s prints out;

Reply:
> not sure if checking restartcount is the best idea, I wonder if the count can be greater than 1 if the pod is evicted from a node and moved to other node or a node dies etc.

In all these cases even if your restart count is > 1 , the pod will not be deleted cause the sub-condition of type and phase will not match, since pod will be in running state.